src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/special/SpecialWikiActivityPageObject.java

```diff
@@ -65,117 +73,101 @@ public void verifyNewWallThreadEntry(String postTitle, String postContent, Strin
     Assertion.assertNotNull(result);
   }
 
-  /**
-   * verifies if there is the wanted edition on WikiActivity, searching through 5 recent editions.
-   */
-  public void verifyRecentEdition(String articleName, String userName) {
+  public Boolean doesLastNRecentEditionsContain(int n, String articleName, String userName) {
     Boolean ifPassed = false;
-    for (int i = 0; i < 4; i++) {
-      ifPassed = ifDetailsPresent(editActivitiesList.get(i), articleName,
-                                  userName);
-      if (ifPassed) {
-        PageObjectLogging.log("verifyRecentEdition",
-                              "WikiActivity module found recent edition that affected '"
-                              + articleName
-                              + "' and was done by user: " + userName, true);
+
+    if(n < editActivitiesList.size()){
+      System.out.println(n);
+      System.out.println(editActivitiesList.size());
+      n = editActivitiesList.size();
+    }
+
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifDetailsPresent(editActivitiesList.get(i), articleName, userName);
+      if (ifPassed == true) {
         break;
       }
     }
-    if (!ifPassed) {
-      PageObjectLogging.log("verifyRecentEdition",
-                            "edition on article  '" + articleName + "' by user: '"
-                            + userName + "', not found", false, driver);
-    }
+
+    return ifPassed;
   }
 
-  /**
-   * verifies if there is the wanted NewPage on WikiActivity, searching through 5 recent editions.
-   */
-  public void verifyRecentNewPage(String articleName, String userName) {
+  public Boolean doesLastNRecentActivitiesContain(int n, String articleName, String userName) {
+
+    if(n < editActivitiesList.size()){
+      n = editActivitiesList.size();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "n". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)